From e0bf97eecfa601e08b9578f00b1000890284241a Mon Sep 17 00:00:00 2001
From: Alexander Kanavin <alex.kanavin@gmail.com>
Date: Tue, 27 Oct 2015 16:02:19 +0200
Subject: [PATCH] FindGObjectIntrospection.cmake: prefix variables obtained
 from pkg-config with PKG_CONFIG_SYSROOT_DIR

See discussion at https://bugs.webkit.org/show_bug.cgi?id=232933 for
reasons why this is not approproiate for upstream submission.

Upstream-Status: Inappropriate [oe-core specific]
Signed-off-by: Alexander Kanavin <alex.kanavin@gmail.com>

---
 Source/cmake/FindGObjectIntrospection.cmake | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/Source/cmake/FindGObjectIntrospection.cmake b/Source/cmake/FindGObjectIntrospection.cmake
index be96814a..ae67b593 100644
--- a/Source/cmake/FindGObjectIntrospection.cmake
+++ b/Source/cmake/FindGObjectIntrospection.cmake
@@ -16,7 +16,6 @@
 # Redistribution and use is allowed according to the terms of the BSD license.
 
 
-
 find_package(PkgConfig QUIET)
 if (PKG_CONFIG_FOUND)
     if (PACKAGE_FIND_VERSION_COUNT GREATER 0)
@@ -30,6 +29,9 @@ if (PKG_CONFIG_FOUND)
         pkg_get_variable(INTROSPECTION_GENERATE gobject-introspection-1.0 g_ir_generate)
         pkg_get_variable(INTROSPECTION_GIRDIR gobject-introspection-1.0 girdir)
         pkg_get_variable(INTROSPECTION_TYPELIBDIR gobject-introspection-1.0 typelibdir)
+        set(INTROSPECTION_SCANNER "$ENV{PKG_CONFIG_SYSROOT_DIR}${INTROSPECTION_SCANNER}")
+        set(INTROSPECTION_COMPILER "$ENV{PKG_CONFIG_SYSROOT_DIR}${INTROSPECTION_COMPILER}")
+        set(INTROSPECTION_GENERATE "$ENV{PKG_CONFIG_SYSROOT_DIR}${INTROSPECTION_GENERATE}")
         set(INTROSPECTION_VERSION "${_pc_gir_VERSION}")
         if (${INTROSPECTION_VERSION} VERSION_GREATER_EQUAL "1.59.1")
             set(INTROSPECTION_HAVE_SOURCES_TOP_DIRS YES)
